src/main/java/com/kruize/service/RecommendationsService.java

```diff
@@ -117,25 +118,41 @@ private JsonObject getApplicationJson(ApplicationRecommendationsImpl application
 
         JsonObject resourcesJson = getResourceJson(applicationRecommendations, application);
 
-        applicationRecommendationJson.add("resources", resourcesJson);
-
-        return applicationRecommendationJson;
+        if (resourcesJson != null)
+        {
+            applicationRecommendationJson.add("resources", resourcesJson);
+            return applicationRecommendationJson;
+        }
+        else
+        {
+            return null;
+        }
     }
 
     private JsonObject getResourceJson(ApplicationRecommendationsImpl applicationRecommendations, String application) throws NoSuchApplicationException
     {
-        JsonObject resourcesJson = new JsonObject();
-
-        JsonObject resourceRequestsJson = new JsonObject();
-        resourceRequestsJson.addProperty("memory", applicationRecommendations.getRssRequests(application) + "M");
-        resourceRequestsJson.addProperty("cpu", applicationRecommendations.getCpuRequests(application));
+        String applicationStatus = applicationRecommendations.getStatus(application);
 
-        JsonObject resourceLimitsJson = new JsonObject();
-        resourceLimitsJson.addProperty("memory", applicationRecommendations.getRssLimits(application) + "M");
-        resourceLimitsJson.addProperty("cpu", applicationRecommendations.getCpuLimit(application));
+        /* If application is still running or idle, or if application is removed,
+            but Kruize has earlier generated recommendations.
+         */
+        if (applicationStatus.equals("running") || applicationStatus.equals("idle")
+                || (applicationRecommendations.getRssRequests(application) != 0))
+        {
+            JsonObject resourcesJson = new JsonObject();
+            JsonObject resourceRequestsJson = new JsonObject();
+            resourceRequestsJson.addProperty("memory", applicationRecommendations.getRssRequests(application) + "M");
+            resourceRequestsJson.addProperty("cpu", applicationRecommendations.getCpuRequests(application));
+
+            JsonObject resourceLimitsJson = new JsonObject();
+            resourceLimitsJson.addProperty("memory", applicationRecommendations.getRssLimits(application) + "M");
+            resourceLimitsJson.addProperty("cpu", applicationRecommendations.getCpuLimit(application));
+
+            resourcesJson.add("requests", resourceRequestsJson);
+            resourcesJson.add("limits", resourceLimitsJson);
+            return resourcesJson;
+        }
 
-        resourcesJson.add("requests", resourceRequestsJson);
-        resourcesJson.add("limits", resourceLimitsJson);
-        return resourcesJson;
+        return null;
```

Comment:
Can we do some logging here to indicate that we are not returning any object and why

Reply:
Added logging statements in the latest commit